CONTRIBUTING.md

```diff
@@ -0,0 +1,28 @@
+# CONTRIBUTING GUIDELINES
+
+NovelRT is a community-driven project. Contribution is aslways welcome, and much appreciated.
+It is also how it'll be developed.
+
+First, please take a moment to review our [code of conduct](CODE_OF_CONDUCT.md).
+
+These guidelines are an attempt at better addressing the huge amount of pending
+issues and pull requests. Please read them closely.
+
+## Submitting Pull Requests
+
+### Getting started
+
+You might know the basics of 
+[contributing on GitHub](https://help.github.com/articles/using-pull-requests) and have a fork
+[properly set up](https://github.com/ohmyzsh/ohmyzsh/wiki/Contribution-Technical-Practices).
+
+You can only create PRs on a dedicated branch.
+
+If you create your own PR, please double check that everything is all right. Also please type what you fixed or added in the PR body,
+[for instance](https://help.github.com/articles/closing-issues-via-commit-messages/)
+
+## Volunteer
+
+Very nice!! :)
```

Comment:
We will need a little more detail here on how contributors can get started 😅 apologies for not having the issue fully written out at the time you picked this up.

More specifically, we want to cover the following:

- How to Get in Touch
-- If they want to ask about a ticket, make a suggestion, show off how they're using NovelRT, if they need assistance with the project or even just chat, they can reach us at Discord 😄 
-- They can also open an issue or start a Github Discussion, too
-- Also, if they have additional (relevant) input on currently open issues, they can feel free to comment as well

- Looking for an Issue
-- Call out here that we do use `help wanted` and `good first issue` labels, but any issue that is unassigned can be picked up.
-- If the issue you're looking at is assigned but hasn't been worked on in a considerable amount of time, reach out to the maintainers about reassignment.

- Contributing to Development (once an Issue has been picked up)
-- This area should link back on how to build NovelRT (from the README)
-- It should also mention our Code Style - we use an automatic formatter `clang-format` to adhere to a codestyle similar to Microsoft's (https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options?view=vs-2017)
-- Also should mention that if they have trouble adhering to the code style, our CI process will produce a patch for them when needed so that they can apply it and make their code adherent.

- Creating a Pull Request
-- This should mention that PRs need to be directed to the `main` branch, but should start from a fork on a different branch.
-- You can specifically call out if they are a first-timer, they can look to the basic links that you provided here.
-- PRs should be tagged with an issue they resolve - PRs without issue need to be briefly discussed with the maintainers first before they will be accepted.
-- Once a PR has been submitted, one or more of the maintainers will review, comment, and/or merge based on if the pull request meets criteria or not.
-- It should be specifically called out that any pull requests created must have their creator agree to our Contributor License Agreement (CLA) before the pull request will be allowed to be merged.

If you'd like a better idea of how the flow of the guide should be, feel free to look here as a reference: https://reactjs.org/docs/how-to-contribute.html

Also, for this piece, you can tag me in comments or reach out to me however if you need additional follow-up or info to help you complete this!